nautobot/utilities/testing/api.py

```diff
@@ -256,6 +256,32 @@ def test_options_returns_expected_choices(self):
 
             self.assertEqual(set(self.choices_fields), field_choices)
 
+        @override_settings(EXEMPT_VIEW_PERMISSIONS=["*"])
+        def test_status_options_returns_expected_choices(self):
+            # Set to self.choices_fields as empty set to compare classes that shouldn't have any choice fields on serializer.
+            if not self.choices_fields:
+                self.choices_fields = set()
+
+            # Don't bother testing if there's no `status` field.
```

Comment:
In the above tests, we decided not to call out the skip, so I'm not sure if we either want to add the `skipTest` to the above tests, or remove it from this test.

I assume using `skipTest` makes the tests look hairy and maybe can cause confusion.

P.S. This test does look really great though. It looks familiar.

Reply:
I opted to `skipTest` for this one because it's only relevant for objects with a `status` field, so there would be no point in testing them otherwise. In the tests above this one, they are more general case so that gating isn't necessary. 

I would argue, however, that for those tests, having each test skip if the test case doesn't have `choice_fields` set might be a good runtime optimization to shave some time off on each iteration if the test doesn't need to be executed, (e.g. skipping setting `is_superuser` and the API call just to compare two empty sets).